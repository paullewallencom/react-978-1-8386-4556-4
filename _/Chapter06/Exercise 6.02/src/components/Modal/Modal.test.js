import React, { useState } from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import { Modal } from "./index";

afterEach(cleanup);

it("renders modal component", () => {
  const { asFragment } = render(<Modal showModal={true}>Some text</Modal>);
  expect(asFragment()).toMatchSnapshot();
});

it("does not show component when showModal is false", () => {
  const { container } = render(
    <Modal showModal={false}>Text not visible</Modal>
  );
  expect(container.firstChild).toBeNull();
});

const ModalDemo = () => {
  const [showModal, setShowModal] = useState(false);
  return (
    <div>
      <Modal
        showModal={showModal}
        toggleModal={() => {
          console.log("clicked");
        }}
      >
        Text not visible
      </Modal>
      <button
        onClick={() => {
          setShowModal(!showModal);
        }}
      >
        Toggle Modal
      </button>
    </div>
  );
};

it("switches modal when button is clicked", () => {
  const { container, getByText } = render(<ModalDemo />);
  expect(container.querySelector(".modal")).toBeNull();
  fireEvent.click(getByText("Toggle Modal"));
  expect(container.querySelector(".modal")).not.toBeNull();
  fireEvent.click(getByText("Toggle Modal"));
  expect(container.querySelector(".modal")).toBeNull();
});
