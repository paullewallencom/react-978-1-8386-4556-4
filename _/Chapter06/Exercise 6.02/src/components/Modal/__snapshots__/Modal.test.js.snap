// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal component 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal_content"
    >
      Some text
      <div>
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
